WordPressKit/WordPressKit/MediaServiceRemoteXMLRPC.m

```diff
@@ -30,13 +30,41 @@ - (void)getMediaWithID:(NSNumber *)mediaID
 - (void)getMediaLibraryWithSuccess:(void (^)(NSArray *))success
                            failure:(void (^)(NSError *))failure
 {
-    NSArray *parameters = [self defaultXMLRPCArguments];
+    NSMutableArray * mediaResult = [[NSMutableArray alloc] init];
+    [self getMediaLibraryStartOffset:0 media:mediaResult success:success failure:failure];
+}
+
+- (void)getMediaLibraryStartOffset:(NSUInteger)offset
+                             media:(NSArray *)media
+                           success:(void (^)(NSArray *))success
+                           failure:(void (^)(NSError *))failure
+{
+    NSInteger pageSize = 100;
+    NSDictionary *filter = @{
+                                @"number": @(pageSize),
+                                @"offset": @(offset)
+                            };
+    NSArray *parameters = [self XMLRPCArgumentsWithExtra:filter];
+
     [self.api callMethod:@"wp.getMediaLibrary"
               parameters:parameters
                  success:^(id responseObject, NSHTTPURLResponse *httpResponse) {
                      NSAssert([responseObject isKindOfClass:[NSArray class]], @"Response should be an array.");
                      if (success) {
-                         success([self remoteMediaFromXMLRPCArray:responseObject]);
+                         NSArray *pageMedia = [self remoteMediaFromXMLRPCArray:responseObject];
+                         NSMutableArray *resultMedia = [NSMutableArray arrayWithArray:media];
```

Comment:
We could remove the mutable array entirely here (and the `if` below) and just do:

`NSArray *resultMedia = [media arrayByAddingObjectsFromArray:pageMedia];`